execution/evm/vm.go

```diff
@@ -733,21 +733,31 @@ func (vm *VM) execute(callState Interface, eventSink EventSink, caller, callee c
 			}))
 			vm.Debugf(" => T:%X D:%X\n", topics, data)
 
-		case CREATE: // 0xF0
+		case CREATE, CREATE2: // 0xF0, 0xFB
 			returnData = nil
+			var salt Word256
 
 			contractValue := stack.PopU64()
 			offset, size := stack.PopBigInt(), stack.PopBigInt()
 			input := memory.Read(offset, size)
 
+			if op == CREATE2 {
+				salt = stack.Pop()
+			}
+
 			// TODO charge for gas to create account _ the code length * GasCreateByte
 			useGasNegative(gas, GasCreateAccount, callState)
 
 			vm.sequence++
 			nonce := make([]byte, txs.HashLength+uint64Length)
 			copy(nonce, vm.tx.Hash())
 			PutUint64BE(nonce[txs.HashLength:], vm.sequence)
-			newAccount := crypto.NewContractAddress(callee, nonce)
+			var newAccount crypto.Address
+			if op == CREATE {
+				newAccount = crypto.NewContractAddress(callee, nonce)
+			} else if op == CREATE2 {
+				newAccount = crypto.NewContractAddress2(callee, nonce, salt, callState.GetCode(callee))
```

Comment:
I was worried that this might not check that this address does not exist but it does via new state interface.